### ISSUE TITLE
JsonException after EurekaHttpClient  SendHeartbeatAsync 

### ISSUE THREAD

**Issue by cecilphillip** [SteeltoeOSS/Steeltoe#441]
## Describe the bug
I have an app using Eureka for discovery. The app registers and discovers services fine. When I turn up the logs to DEBUG on Steeltoe logs I'm getting a ton of JsonExceptions whenever the eureka client calls SendHeartbeatAsync in the background. It's not affecting the app but it is slightly annoying. Is there anything I can do to stop that (other than stop reading the debug logs 🙂). I'm wondering if it's related to those `Missing span context` messges.

## Steps to reproduce 
1. Set Steeltoe logs to Debug
2. Configured service discovery to use Eureka

## Environment (please complete the following information):
 - Platform: Tye / Docker
 - OS: MacOS
 - .NET Version: NET Core 3.1
 - Steeltoe Version. 3.0
 - Any other library versions to note: StackExchange.Redis, Rabbitmq.Client

## Additional context or links
Here are the logs
```
)
[website_08225f34-a]: at Steeltoe.Discovery.Eureka.Transport.EurekaHttpClient.SendHeartBeatAsyncInternal(String appName, String id, InstanceInfo info, InstanceStatus overriddenStatus)
[website_08225f34-a]: [DBG]: 15-09-2020:23:12:24 Steeltoe.Discovery.Eureka.Transport.EurekaHttpClient SendHeartbeatAsync http://localhost:8761/eureka/apps/CONTOSO-WEBSITE/apex-macbook-2018.local:contoso-website:5000?status=UP&lastDirtyTimestamp=1600225884183, status: OK, instanceInfo: null, retry: 0
[website_08225f34-a]: [DBG]: 15-09-2020:23:12:24 Steeltoe.Discovery.Eureka.DiscoveryClient Renew CONTOSO-WEBSITE/apex-macbook-2018.local:contoso-website:5000 returned: OK
[website_08225f34-a]: [DBG]: 15-09-2020:23:12:24 Steeltoe.Discovery.Eureka.Transport.EurekaHttpClient DoGetApplicationsAsync http://localhost:8761/eureka/apps/, status: OK, applications: Applications[Application[Name=CONTOSO-WEBSITE,Instances=Instance[InstanceId=apex-macbook-2018.local:contoso-website:5000,HostName=localhost,IpAddr=localhost,Status=UP,IsUnsecurePortEnabled=False,Port=5000,IsSecurePortEnabled=False,SecurePort=443,VipAddress=contoso-website,SecureVipAddress=contoso-website,ActionType=ADDED],]Application[Name=CONTOSO-PRODUCTSAPI,Instances=Instance[InstanceId=apex-macbook-2018.local:contoso-productsapi:5005,HostName=localhost,IpAddr=localhost,Status=UP,IsUnsecurePortEnabled=False,Port=5005,IsSecurePortEnabled=False,SecurePort=443,VipAddress=contoso-productsapi,SecureVipAddress=contoso-productsapi,ActionType=ADDED],]], retry: 0
[website_08225f34-a]: [DBG]: 15-09-2020:23:12:24 Steeltoe.Discovery.Eureka.DiscoveryClient FetchFullRegistry returned: OK, Applications[Application[Name=CONTOSO-WEBSITE,Instances=Instance[InstanceId=apex-macbook-2018.local:contoso-website:5000,HostName=localhost,IpAddr=localhost,Status=UP,IsUnsecurePortEnabled=False,Port=5000,IsSecurePortEnabled=False,SecurePort=443,VipAddress=contoso-website,SecureVipAddress=contoso-website,ActionType=ADDED],]Application[Name=CONTOSO-PRODUCTSAPI,Instances=Instance[InstanceId=apex-macbook-2018.local:contoso-productsapi:5005,HostName=localhost,IpAddr=localhost,Status=UP,IsUnsecurePortEnabled=False,Port=5005,IsSecurePortEnabled=False,SecurePort=443,VipAddress=contoso-productsapi,SecureVipAddress=contoso-productsapi,ActionType=ADDED],]]
[website_08225f34-a]: [DBG]: 15-09-2020:23:12:24 Steeltoe.Discovery.Eureka.DiscoveryClient FetchRegistry succeeded
[website_08225f34-a]: [DBG]: 15-09-2020:23:12:24 Steeltoe.Management.Tracing.Observer.HttpClientCoreObserver HandleStartEvent: Ignoring path: /api/v2/spans
[website_08225f34-a]: [DBG]: 15-09-2020:23:12:24 Steeltoe.Management.Tracing.Observer.HttpClientCoreObserver HandleStopEvent: Missing span context
[website_08225f34-a]: [DBG]: 15-09-2020:23:12:29 Steeltoe.Management.Tracing.Observer.HttpClientCoreObserver HandleStartEvent: Ignoring path: /api/v2/spans
[website_08225f34-a]: [DBG]: 15-09-2020:23:12:29 Steeltoe.Management.Tracing.Observer.HttpClientCoreObserver HandleStopEvent: Missing span context
[website_08225f34-a]: [DBG]: 15-09-2020:23:12:34 Steeltoe.Management.Tracing.Observer.HttpClientCoreObserver HandleStartEvent: Ignoring path: /api/v2/spans
[website_08225f34-a]: [DBG]: 15-09-2020:23:12:34 Steeltoe.Management.Tracing.Observer.HttpClientCoreObserver HandleStopEvent: Missing span context
[website_08225f34-a]: [DBG]: 15-09-2020:23:12:39 Steeltoe.Management.Tracing.Observer.HttpClientCoreObserver HandleStartEvent: Ignoring path: /api/v2/spans
[website_08225f34-a]: [DBG]: 15-09-2020:23:12:39 Steeltoe.Management.Tracing.Observer.HttpClientCoreObserver HandleStopEvent: Missing span context
[website_08225f34-a]: [DBG]: 15-09-2020:23:12:44 Steeltoe.Management.Tracing.Observer.HttpClientCoreObserver HandleStartEvent: Ignoring path: /api/v2/spans
[website_08225f34-a]: [DBG]: 15-09-2020:23:12:44 Steeltoe.Management.Tracing.Observer.HttpClientCoreObserver HandleStopEvent: Missing span context
[website_08225f34-a]: [DBG]: 15-09-2020:23:12:49 Steeltoe.Management.Tracing.Observer.HttpClientCoreObserver HandleStartEvent: Ignoring path: /api/v2/spans
[website_08225f34-a]: [DBG]: 15-09-2020:23:12:49 Steeltoe.Management.Tracing.Observer.HttpClientCoreObserver HandleStopEvent: Missing span context
[website_08225f34-a]: [INF]: 15-09-2020:23:12:54 Steeltoe.Discovery.Eureka.Transport.EurekaHttpClient Response could not be deserialized
[website_08225f34-a]: System.Text.Json.JsonException: The input does not contain any JSON tokens. Expected the input to start with a valid JSON token, when isFinalBlock is true. Path: $ | LineNumber: 0 | BytePositionInLine: 0.
[website_08225f34-a]: ---> System.Text.Json.JsonReaderException: The input does not contain any JSON tokens. Expected the input to start with a valid JSON token, when isFinalBlock is true. LineNumber: 0 | BytePositionInLine: 0.
[website_08225f34-a]: at System.Text.Json.ThrowHelper.ThrowJsonReaderException(Utf8JsonReader& json, ExceptionResource resource, Byte nextByte, ReadOnlySpan`1 bytes)
[website_08225f34-a]: at System.Text.Json.Utf8JsonReader.Read()
[website_08225f34-a]: at System.Text.Json.JsonSerializer.ReadCore(JsonSerializerOptions options, Utf8JsonReader& reader, ReadStack& readStack)
[website_08225f34-a]: --- End of inner exception stack trace ---
[website_08225f34-a]: at System.Text.Json.ThrowHelper.ReThrowWithPath(ReadStack& readStack, JsonReaderException ex)
[website_08225f34-a]: at System.Text.Json.JsonSerializer.ReadCore(JsonSerializerOptions options, Utf8JsonReader& reader, ReadStack& readStack)
[website_08225f34-a]: at System.Text.Json.JsonSerializer.ReadCore(JsonReaderState& readerState, Boolean isFinalBlock, ReadOnlySpan`1 buffer, JsonSerializerOptions options, ReadStack& readStack)
[website_08225f34-a]: at System.Text.Json.JsonSerializer.ReadAsync[TValue](Stream utf8Json, Type returnType, JsonSerializerOptions options, CancellationToken cancellationToken)
[website_08225f34-a]: at System.Net.Http.Json.HttpContentJsonExtensions.ReadFromJsonAsyncCore[T](HttpContent content, Encoding sourceEncoding, JsonSerializerOptions options, CancellationToken cancellationToken)
[website_08225f34-a]: at Steeltoe.Discovery.Eureka.Transport.EurekaHttpClient.SendHeartBeatAsyncInternal(String appName, String id, InstanceInfo info, InstanceStatus overriddenStatus)
[website_08225f34-a]: [DBG]: 15-09-2020:23:12:54 Steeltoe.Discovery.Eureka.Transport.EurekaHttpClient SendHeartbeatAsync http://localhost:8761/eureka/apps/CONTOSO-WEBSITE/apex-macbook-2018.local:contoso-website:5000?status=UP&lastDirtyTimestamp=1600225884183, status: OK, instanceInfo: null, retry: 0
[website_08225f34-a]: [DBG]: 15-09-2020:23:12:54 Steeltoe.Discovery.Eureka.DiscoveryClient Renew CONTOSO-WEBSITE/apex-macbook-2018.local:contoso-website:5000 returned: OK
[website_08225f34-a]: [DBG]: 15-09-2020:23:12:54 Steeltoe.Discovery.Eureka.Transport.EurekaHttpClient DoGetApplicationsAsync http://localhost:8761/eureka/apps/delta, status: OK, applications: Applications[Application[Name=CONTOSO-WEBSITE,Instances=Instance[InstanceId=apex-macbook-2018.local:contoso-website:5000,HostName=localhost,IpAddr=localhost,Status=UP,IsUnsecurePortEnabled=False,Port=5000,IsSecurePortEnabled=False,SecurePort=443,VipAddress=contoso-website,SecureVipAddress=contoso-website,ActionType=ADDED],]Application[Name=CONTOSO-PRODUCTSAPI,Instances=Instance[InstanceId=apex-macbook-2018.local:contoso-productsapi:5005,HostName=localhost,IpAddr=localhost,Status=UP,IsUnsecurePortEnabled=False,Port=5005,IsSecurePortEnabled=False,SecurePort=443,VipAddress=contoso-productsapi,SecureVipAddress=contoso-productsapi,ActionType=ADDED],]Application[Name=CONTOSO-CHECKOUTPROCESSOR,Instances=Instance[InstanceId=apex-macbook-2018.local:contoso-checkoutprocessor:80,HostName=localhost,IpAddr=localhost,Status=UP,IsUnsecurePortEnabled=False,Port=80,IsSecurePortEnabled=False,SecurePort=443,VipAddress=contoso-checkoutprocessor,SecureVipAddress=contoso-checkoutprocessor,ActionType=ADDED],]], retry: 0
[website_08225f34-a]: [DBG]: 15-09-2020:23:12:54 Steeltoe.Discovery.Eureka.DiscoveryClient FetchRegistryDelta returned: OK
[website_08225f34-a]: [DBG]: 15-09-2020:23:12:54 Steeltoe.Discovery.Eureka.DiscoveryClient FetchRegistry succeeded
```
Here's the relevant part of my settings file
```json
  "Eureka": {
    "Client": {
      "ServiceUrl": "http://localhost:8761/eureka",
      "validateCertificates": false
    },
    "Instance": {
      "AppName": "contoso-website",
      "Port": 5000,
      "IpAddress": "localhost",
      "PreferIpAddress": true
    }
  },
  "Management": {
    "tracing": {
      "Name": "website",
      "AlwaysSample": true,
      "exporter": {
        "zipkin": {
          "ServiceName": "website",
          "Endpoint": "http://localhost:9411/api/v2/spans",
          "ValidateCertificates": false
        }
      }
    }
  },
```

**Comment by TimHess** [comment]
Thanks for logging the issue @cecilphillip - this is fixed in #434, here: https://github.com/SteeltoeOSS/Steeltoe/pull/434/files#diff-e4612c6eca67ad1c6fa0a7602b9f3873R212

For what it's worth, the response body from Eureka is empty where Steeltoe was expecting something. The log entry is made within a catch block and that response object does not appear to have been used for anything significant, so it is a message that can be ignored.  I wouldn't think the span context messages would be related, but maybe @hananiel can offer another opinion

**Comment by hananiel** [comment]
I don't think span context messages are related either.
